lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,29 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
```

Comment:
@zachthompson Everything apart from string equality.

For example, if you wanted to get all the IAs made after a certain date you could do 

```perl
my $ias = filter_ias({creation_date => \&less_than_two_years_old});
```